docs/diagrams/wallet-lifecycle/wallet-lifecycle-3-responding-to-moving-funds-state.tikz

```diff
@@ -0,0 +1,153 @@
+   start/.style = {base, fill=blue!30},
+  revert/.style = {base, fill=red!30},
+  action/.style = {base, fill=black!5},
+   check/.style = {diamond, aspect=1.2, draw=black, fill=black!5,
+                   minimum width=1cm, align=center},
+   note1/.style = {rectangle, draw, dashed, fill=green!10,
+                   align=flush left, text width=5cm},
+   note2/.style = {rectangle, draw, dashed, fill=cyan!10,
+                   align=flush left, text width=5cm},
+              > = Triangle,                  % defines the arrow tip style when `->` is used
+           font = \sffamily,
+]
+
+% RESET MOVING FUNDS TIMEOUT
+% Nodes
+\node[start]                                                       (rmft_start)   {\textbf{Reset moving funds timeout}};
+\node[check, below=of rmft_start]                                  (rmft_check1)  {Is wallet\\ MOVING\_FUNDS?};
+\node[check, below right=of rmft_check1]                           (rmft_check2)  {Has the\\ moving funds\\ commitment been already\\ submitted for\\ the wallet?};
+\node[check, below left=of rmft_check2]                            (rmft_check3)  {Is \# of live wallets equal 0?};
+\node[revert, below right=of rmft_check2]                          (rmft_revert1) {Transaction reverted};
+\node[check, below right=of rmft_check3]                           (rmft_check4)  {Has at least\\ moving\_funds\_timeout\_reset\_delay\\ passed since moving\_funds\_timeout\\ was reset for the wallet?};
+\node[revert, below left=of rmft_check4, xshift=-3cm]                           (rmft_revert2) {Transaction reverted};
+\node[action, below right=of rmft_check4]                          (rmft_action1) {Reset the moving\_funds\_timeout};
+\node[action, below=of rmft_action1]                               (rmft_action2) {Notification sent};
+\node[note1, right=of rmft_start.south east, anchor=south west]    (rmft_note1)   {Can be successfully called for wallets:\\ 
+                                                                                   \textbf{- MOVING\_FUNDS}};
+\node[note2, above=of rmft_note1.north, anchor=south, yshift=-8mm] (rmft_note2)   {Sender incentivized by:\\ \textbf{Reimbursement, if maintainer}};
+
+% Arrows
+\draw[->] (rmft_start)   -- (rmft_check1);
+\draw[->] (rmft_check1)  -| (rmft_revert2) \no;
+\draw[->] (rmft_check1)  -| (rmft_check2)  \yes;
+\draw[->] (rmft_check2)  -| (rmft_check3)  \no;
+\draw[->] (rmft_check2)  -| (rmft_revert1) \yes;
+\draw[->] (rmft_check3)  -| (rmft_revert2) \no;
+\draw[->] (rmft_check3)  -| (rmft_check4)  \yes;
+\draw[->] (rmft_check4)  -| (rmft_revert2) \no;
+\draw[->] (rmft_check4)  -| (rmft_action1) \yes;
+\draw[->] (rmft_action1) -- (rmft_action2);
+
+% MOVING FUNDS COMMITMENT
+% Nodes
+\node[start, right=of rmft_start, xshift=12cm]                    (mfc_start)   {Somebody sends \textbf{moving funds commitment}};
+\node[check, below=of mfc_start]                                  (mfc_check1)  {Is the\\ source wallet in\\ MOVING\_FUNDS\\ state?};
```

Comment:
Maybe

> Is the source wallet MOVING_FUNDS?

On other diagrams we say *Is wallet MOVING_FUNDS*, *Is wallet LIVE* and so on.